src/Microsoft.AspNet.Razor/Parser/TokenizerBackedParser.Helpers.cs

```diff
@@ -147,12 +149,12 @@ protected internal bool Balance(BalancingModes mode, TSymbolType left, TSymbolTy
 
         protected internal bool NextIs(TSymbolType type)
         {
-            return NextIs(sym => sym != null && Equals(type, sym.Type));
+            return NextIs(sym => sym != null && SymbolTypeEquals(type, sym.Type));
         }
 
         protected internal bool NextIs(params TSymbolType[] types)
         {
-            return NextIs(sym => sym != null && types.Any(t => Equals(t, sym.Type)));
+            return NextIs(sym => sym != null && types.Any(t => SymbolTypeEquals(t, sym.Type)));
```

Comment:
Is this called? We might want to clean this up to be more allocation friendly while here